src/test/java/com/egov/business-services/pretests/egfMasterChartOfAccount.feature

```diff
@@ -0,0 +1,193 @@
+    Then status 200
+    And def chartOfAccountSearchResponseHeader = responseHeaders
+    And def chartOfAccountSearchResponseBody = response
+    * print chartOfAccountSearchResponseBody
+
+@searchWithTenantIdSuccessfully
+Scenario: Searching chart of accounts with TenantId through API call
+  * def chartOfAccountSearcheParam = 
+    """
+    {
+     tenantId: '#(tenantId)'
+    }
+    """ 
+    Given url charOfAccountSearch
+    * print charOfAccountSearch 
+    And params chartOfAccountSearcheParam
+    * print chartOfAccountSearcheParam
+    And request chartOfAccountSearchPayload
+    * print chartOfAccountSearchPayload
+    When method post
+    Then status 200
+    And def chartOfAccountSearchResponseHeader = responseHeaders
+    And def chartOfAccountSearchResponseBody = response
+    * print chartOfAccountSearchResponseBody
+
+@searchAccountError
+Scenario: Searching chart of accounts and check for error through API call
+  * def chartOfAccountSearcheParam = 
+    """
+    {
+     id: '#(id)',
+     glcode: '#(glcode)',
+     classification: '#(classification)',
+     name: '#(name)',
+     type: '#(type)',
+     isActiveForPosting: '#(isActiveForPosting)'
+    }
+    """ 
+    Given url charOfAccountSearch
+    * print charOfAccountSearch 
+    And params chartOfAccountSearcheParam
+    * print chartOfAccountSearcheParam
+    And request chartOfAccountSearchPayload
+    * print chartOfAccountSearchPayload
+    When method post
+    Then status 400
+    And def chartOfAccountSearchResponseHeader = responseHeaders
+    And def chartOfAccountSearchResponseBody = response
+    * print chartOfAccountSearchResponseBody
+
+# Chart of Account Update
+@updateAccountSuccessfully
+Scenario: Updating chart of accounts through API call
+    * def chartOfAccountUpdateParam = 
+    """
+    {
+     tenantId: '#(tenantId)'
+    }
+    """ 
+     Given url chartOfAccountUpdate
+     * print chartOfAccountUpdate 
+     And params chartOfAccountUpdateParam
+     * print chartOfAccountUpdateParam
+     And request chartOfAccountUpdatePayload
+     * print chartOfAccountUpdatePayload
+     When method post
+     Then status 201
+     And def chartOfAccountUpdateResponseHeader = responseHeaders
+     And def chartOfAccountUpdateResponseBody = response
+     * print chartOfAccountUpdateResponseBody
+
+@updateAccountError
+Scenario: Updating chart of accounts and check for error through API call
+    * def chartOfAccountUpdateParam = 
+    """
+    {
+     tenantId: '#(tenantId)'
+    }
+    """ 
+     Given url chartOfAccountUpdate
+     * print chartOfAccountUpdate 
+     And params chartOfAccountUpdateParam
+     * print chartOfAccountUpdateParam
+     And request chartOfAccountUpdatePayload
+     * print chartOfAccountUpdatePayload
+     When method post
+     Then status 400
+     And def chartOfAccountUpdateResponseHeader = responseHeaders
+     And def chartOfAccountUpdateResponseBody = response
+     * print chartOfAccountUpdateResponseBody
```

Comment:
And as per our last discussion, these pretest scenarios can be optimized. Instead of creating multiple pretest scenarios for almost all of the tests, please try to merge multiple scenarios in one pretest scenario. Until some new URL or steps are encountered try to reuse already exists pretest, for status code validation you may use the `||` operator. Like: `Then status 200 || 400 `

Reply:
Here pretests are different as per service. So I am keeping all the pretests.